src/plugins/kibana_usage_collection/server/collectors/event_loop_delays/saved_objects.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type {
+  SavedObjectAttributes,
+  SavedObjectsServiceSetup,
+  ISavedObjectsRepository,
+} from 'kibana/server';
+import moment from 'moment';
+import type { IntervalHistogram } from './event_loop_delays';
+
+export const SAVED_OBJECTS_DAILY_TYPE = 'event_loop_delays_daily';
+
+export interface EventLoopDelaysDaily extends SavedObjectAttributes, IntervalHistogram {
+  processId: number;
+}
+
+export function registerSavedObjectTypes(registerType: SavedObjectsServiceSetup['registerType']) {
+  registerType({
+    name: SAVED_OBJECTS_DAILY_TYPE,
+    hidden: false,
+    namespaceType: 'agnostic',
+    mappings: {
+      dynamic: false,
+      properties: {
+        // This type requires `timestamp` to be indexed so we can use it when rolling up totals (timestamp < now-90d)
+        timestamp: { type: 'date' },
+      },
+    },
+  });
+}
+
+export function serializeSavedObjectId({ date, pid }: { date: moment.MomentInput; pid: number }) {
+  const formattedDate = moment(date).format('DDMMYYYY');
```

Comment:
I prefer using `moment` for dates especially when it comes to formatting etc. `.format('DDMMYYYY')` is a lot more readable and less error prone than the suggestion above IMO.

Reply:
> is a lot more readable

I do agree it's more readable, I am just not sure that using `moment` for such simple functionality justifies the performance penalty. Yeah, it's not significant per se, but it's a road to death by a thousand cuts.

Reply:
I'm generally against focusing on micro optimizations. There are a lot of bigger performance bottlenecks in an i/o project like kibana.

I dont think kibana would ever suffer in performance because JS can execute a `moment` function 200k times per second while a native date would execute 800k times. I'd always favor readablity and maintanablity in these cases. Especially since we have `moment` integrated in every place where we use dates even in eui components.